controllers/metal3.io/hostfirmwaresettings_controller.go

```diff
@@ -205,13 +211,16 @@ func (r *HostFirmwareSettingsReconciler) updateStatus(info *rInfo, settings meta
 			continue
 		}
 
-		info.hfs.Status.Settings[k] = v
+		newStatus.Settings[k] = v
+	}
+	if !reflect.DeepEqual(info.hfs.Status.Settings, newStatus.Settings) {
+		dirty = true
```

Comment:
nit: `dirty` could directly be set in one line here, by comparing the schema or the the settings

Reply:
Good point, combined to one line